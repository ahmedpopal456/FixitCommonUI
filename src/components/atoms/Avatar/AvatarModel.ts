import { TextStyle } from 'react-native';
import colors from '../../../theme/colors';

export interface AvatarProps {
    style?: TextStyle & React.CSSProperties;
    /** Any color from the theme's color palette */
    color?: keyof typeof colors
    /** The name of the icon */
    icon?:string;
    /** Title of the avatar */
    title?:string;
    /** Height of avatar */
    height?:number;
    /** Width of avatar */
    width?:number;
    /** Image of the avatar */
    image?:string;
  }
