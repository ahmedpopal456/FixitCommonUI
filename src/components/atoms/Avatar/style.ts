import styled from "styled-components/native";
import colors from '../../../theme/colors';

/** PROPS */
interface AvatarProps {
    /**  To override default style */
    style?: any;
    /** Any color from the theme's color palette */
    color?: keyof typeof colors;
    /** Border width of the Text box */
    borderWidth?: number;
    /** Border radius of the Text box (square or round)*/
    borderRadius?: number;
    /**  The title of the Text box */
    title?:string;
    /** To pass custom icon */
    icon?:any;
    backgroundColor?:keyof typeof colors;
    /** Height of the Text box */
    height?: number;
    /** Width of the Text box */
    width?: number;
    /** Placeholder for the Text box */
    placeholder?:string;
    image?:any;
}

export const AvatarWrapper = styled.View`
${(props: AvatarProps) =>
    props.height &&
    `
    height:${props.height};
 `}
 ${(props: AvatarProps) =>
    props.width &&
    `
    width:${props.width};
 `}
    borderRadius:50;
    align-items:center;
    ${(props: AvatarProps) =>
        props.backgroundColor &&
        `
        backgroundColor:${props.backgroundColor};
     `}
    justify-content:center;
    margin:10px;
`;

export const IconWrapper = styled.View`
    align-self:center;
`;

export const TextWrapper =styled.Text`
    align-self:center;
    font-size:20;
    font-weight:bold;
    font-family:Cochin;
    color:white;
`;

export const ImageWrapper = styled.ImageBackground`
    flex:1;
    width:50;
    height:50;
    borderRadius:50;
    overflow:hidden;
    align-self:center;
`;