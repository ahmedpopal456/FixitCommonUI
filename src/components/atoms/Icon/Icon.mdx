---
name: Icon
menu: Atoms
---

import { Props, Playground } from "docz";
import { Icon } from "./Icon.tsx";

# Icon

A Icon is a clickable view to carry out actions such as clearing user input or submitting the form.

# Usage

```tsx
import { Icon } from "fixit-mobile-components";

<Icon />;
```

## Props

<Props of={Icon}></Props>

## Default

<Playground>
  <Icon name="emoticon-kiss" />
</Playground>

## Size

The default size is 24.

<Playground>
  <Icon name="format-size" size={10} />
  <Icon name="format-size" size={20} />
  <Icon name="format-size" size={30} />
  <Icon name="format-size" size={40} />
  <Icon name="format-size" size={50} />
  <Icon name="format-size" size={60} />
  <Icon name="format-size" size={70} />
</Playground>

## Color

The colors must be from the Fixit color pallette.

<Playground>
  <Icon name="palette-swatch" color="primary" />
  <Icon name="palette-swatch" color="accent" />
  <Icon name="palette-swatch" color="grey" />
  <Icon name="palette-swatch" color="light" />
  <Icon name="palette-swatch" color="dark" />
  <Icon name="palette-swatch" color="red" />
  <Icon name="palette-swatch" color="orange" />
  <Icon name="palette-swatch" color="yellow" />
  <Icon name="palette-swatch" color="blue" />
  <Icon name="palette-swatch" color="green" />
  <Icon name="palette-swatch" color="notification" />
  <Icon name="palette-swatch" color="error" />
  <Icon name="palette-swatch" color="success" />
</Playground>

## Icon Library

<Playground>
  <Icon library="Entypo" name="github" />
</Playground>
