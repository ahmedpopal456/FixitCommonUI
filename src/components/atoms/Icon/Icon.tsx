import React from "react";

import colors from "../../../theme/colors";

import { IconLibraries } from "./IconLibraries";
import { IconProps } from "./icon-props.model";

/** Building the Icon */
export const Icon: React.FC<IconProps> = (props) => {
  const VectorIcon = IconLibraries[props.library!];
  const color: string = colors[props.color!];

  const childProps = { ...props };
  delete childProps.color;

  return <VectorIcon {...childProps} name={props.name} color={color} />;
};

Icon.defaultProps = {
  library: "MaterialCommunityIcons",
  color: "primary",
  size: 24,
};

export default Icon;
