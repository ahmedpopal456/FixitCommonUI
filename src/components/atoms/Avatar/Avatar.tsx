import React from "react";
import { AvatarProps } from "./AvatarModel";
import {AvatarWrapper, IconWrapper, TextWrapper, ImageWrapper } from './style'

export const Avatar: React.FC<AvatarProps> = (avatarProps: AvatarProps) => {
    return (
        <AvatarWrapper {...avatarProps}>
          <TextWrapper>{avatarProps.title}</TextWrapper>
          <IconWrapper>{avatarProps.icon}</IconWrapper>
          <ImageBackground {...avatarProps}/>
        </AvatarWrapper>
    );
  };

export const ImageBackground: React.FC<AvatarProps> = (avatarProps: AvatarProps) => {
    if(avatarProps.image != null){
        return (
                <ImageWrapper {...avatarProps} source={{uri:avatarProps.image}}/>
            );
    }
    return null;
  };

  Avatar.defaultProps={
    style:{
      height:50,
      width:50,
      backgroundColor:'grey',
      color:'white'
    }
  };
