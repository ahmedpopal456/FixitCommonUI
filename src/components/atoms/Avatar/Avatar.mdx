---
name: Avatar
menu: Atoms
---

import { Props, Playground } from 'docz';
import { View, ImageBackground } from 'react-native';
import { DoczContainer } from '../../../utils/DoczContainer.tsx'
import { Avatar } from './Avatar.tsx';
import { Icon } from '../Icon/Icon.tsx';

# Avatar
A Button is a clickable view to carry out actions such as clearing user input or submitting the form.

# Usage
```tsx
import { Avatar } from 'fixit-common-ui';
```

## Props

<Props of={Avatar}></Props>

## Title
<Playground>
    <DoczContainer>
        <Avatar title="FI"/>
        <Avatar title="AS" backgroundColor="black" style={{width:60, height:60}}/>
        <Avatar title="JD" backgroundColor="#FFD14A"style={{width:70, height:70}}/>
    </DoczContainer>
</Playground>

## Icon
<Playground>
    <DoczContainer>
        <Avatar icon={<Icon library="AntDesign" name="user" color="white" size={30}/>}/>
        <Avatar backgroundColor="black" style={{width:60, height:60}} icon={<Icon library="AntDesign" name="user" color="white" size={30}/>}/>
        <Avatar backgroundColor="#FFD14A" style={{width:70, height:70}} icon={<Icon library="AntDesign" name="user" color="white" size={30}/>}/>
    </DoczContainer>
</Playground>

## Image
<Playground>
    <DoczContainer>
        <Avatar image={"https://reactjs.org/logo-og.png"}/>
        <Avatar style={{width:60, height:60}} image={"https://reactjs.org/logo-og.png"}/>
        <Avatar style={{width:70, height:70}} image={"https://reactjs.org/logo-og.png"}/>
    </DoczContainer>
</Playground>