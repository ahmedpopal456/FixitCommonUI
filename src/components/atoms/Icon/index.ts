import { Icon } from './Icon';
import { IconLibraries } from './IconLibraries';

export { Icon, IconLibraries };
