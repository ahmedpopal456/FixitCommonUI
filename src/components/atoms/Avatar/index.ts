import { AvatarBadge } from './Avatar';
export { AvatarBadge};