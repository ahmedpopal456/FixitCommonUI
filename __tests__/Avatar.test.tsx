import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Avatar } from '../src/components/atoms/Avatar';
import { Icon } from '../src/components/atoms/Icon';

// Note: test renderer must be required after react-native.

it('matches title snapshot', () => {
  const textInput = renderer.create(<Avatar title="Name"/>);
  expect(textInput.toJSON()).toMatchSnapshot();
});

it('matches icon snapshot', () => {
  const textInput = renderer.create(<Avatar icon={<Icon library="AntDesign" name="user" color="white" size={30}/>}/>);
  expect(textInput.toJSON()).toMatchSnapshot();
});

it('matches image snapshot', () => {
  const textInput = renderer.create(<Avatar image={'https://reactjs.org/logo-og.png'}/>);
  expect(textInput.toJSON()).toMatchSnapshot();
});
