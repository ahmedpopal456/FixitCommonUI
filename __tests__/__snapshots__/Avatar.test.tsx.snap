// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches icon snapshot 1`] = `
<View
  icon={
    <Icon
      color="white"
      library="AntDesign"
      name="user"
      size={30}
    />
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
      Object {
        "backgroundColor": "grey",
        "color": "white",
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontFamily": "Cochin",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      library="AntDesign"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î™ª
    </Text>
  </View>
</View>
`;

exports[`matches image snapshot 1`] = `
<View
  image="https://reactjs.org/logo-og.png"
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
      Object {
        "backgroundColor": "grey",
        "color": "white",
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontFamily": "Cochin",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
        },
      ]
    }
  />
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "borderRadius": 50,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "grey",
          "color": "white",
          "height": 50,
          "width": 50,
        },
      ]
    }
  >
    <Image
      image="https://reactjs.org/logo-og.png"
      source={
        Object {
          "uri": "https://reactjs.org/logo-og.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`matches title snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
      Object {
        "backgroundColor": "grey",
        "color": "white",
        "height": 50,
        "width": 50,
      },
    ]
  }
  title="Name"
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontFamily": "Cochin",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Name
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
        },
      ]
    }
  />
</View>
`;
